src/sql/src/session/vars.rs

```diff
@@ -860,6 +860,15 @@ const PG_REPLICATION_TCP_USER_TIMEOUT: ServerVar<Duration> = ServerVar {
     internal: true,
 };
 
+/// Sets the `statement_timeout` value to use during the snapshotting phase of
+/// PG sources.
+const PG_SNAPSHOT_STATEMENT_TIMEOUT: ServerVar<Duration> = ServerVar {
+    name: UncasedStr::new("pg_snapshot_statement_timeout"),
```

Comment:
Preference to spell out more clearly that this is about PostgreSQL sources and not other PG-like things (e.g., pgwire, the SQL interface generally, whatever).

```suggestion
    name: UncasedStr::new("postgres_source_snapshot_statement_timeout"),
```

Reply:
Ah, crap, just saw that we introduced `pg_replication_*` variables already. Oh well. Scratch my suggestion. Consistency with what we have is better.

Reply:
@benesch The `pg_replication_*` variables will eventually go away in favor of being applied more generally to all PG client connections. Let me see if I can bundle that work into this and we can make the names less ambiguous overall.